Libraries/PeriphDrivers/Source/LP/lp_me14.c

```diff
@@ -574,3 +549,226 @@ void MXC_LP_ICacheXIPLightSleepEnable(void)
+    SystemCoreClockUpdate();
+}
+
+static void save_preDeepSleep_state(void)
+{
+    /* Save ICC state */
+    icc0_state = 0;
+    if (MXC_ICC0->cache_ctrl & MXC_F_ICC_CACHE_CTRL_EN)
+        icc0_state |= (1 << CTRL_POS);
+    if (MXC_PWRSEQ->lpmemsd & MXC_F_PWRSEQ_LPMEMSD_ICACHESD)
+        icc0_state |= (1 << ICSD_POS);
+    icc1_state = 0;
+    if (MXC_ICC1->cache_ctrl & MXC_F_ICC_CACHE_CTRL_EN)
+        icc1_state |= (1 << CTRL_POS);
+    if (MXC_PWRSEQ->lpmemsd & MXC_F_PWRSEQ_LPMEMSD_IC1SD)
+        icc1_state |= (1 << ICSD_POS);
+
+    /* Save CLKCN state */
+    clkcn_state = MXC_GCR->clkcn;
+}
+
+static void restore_preDeepSleep_state(void)
+{
+    /* Restore CLKCN state */
+    MXC_GCR->clkcn = clkcn_state;
+    /* Wait for the switch to occur */
+    while (!(MXC_GCR->clkcn & MXC_F_GCR_CLKCN_CKRDY)) {}
+    SystemCoreClockUpdate();
+
+    /* Restore ICC1 state */
+    if (icc1_state & (1 << ICSD_POS)) {
+        /* ICC power down. Do not restore Enable state. */
+        MXC_PWRSEQ->lpmemsd |= MXC_F_PWRSEQ_LPMEMSD_IC1SD;
+    } else {
+        /* ICC power up */
+        MXC_PWRSEQ->lpmemsd &= ~MXC_F_PWRSEQ_LPMEMSD_IC1SD;
+        /* Enable */
+        if (icc1_state & (1 << CTRL_POS))
+            MXC_ICC_RevA_Enable((mxc_icc_reva_regs_t *)MXC_ICC1);
+    }
+
+    /* Restore ICC0 state */
+    if (icc0_state & (1 << ICSD_POS)) {
+        /* ICC power down. Do not restore Enable state. */
+        MXC_PWRSEQ->lpmemsd |= MXC_F_PWRSEQ_LPMEMSD_ICACHESD;
+    } else {
+        /* ICC power up */
+        MXC_PWRSEQ->lpmemsd &= ~MXC_F_PWRSEQ_LPMEMSD_ICACHESD;
+        /* Enable */
+        if (icc0_state & (1 << CTRL_POS))
+            MXC_ICC_RevA_Enable((mxc_icc_reva_regs_t *)MXC_ICC0);
+    }
+}
+
+void MXC_LP_EnterDeepSleepMode(void)
+{
+    save_preDeepSleep_state();
+
+    /*void prepForDeepSleep(void)*/
+    {
+        MXC_ICC_Disable();
+        MXC_LP_ICache0Shutdown();
+
+        /* Shutdown unused power domains */
+        MXC_PWRSEQ->lpcn |= MXC_F_PWRSEQ_LPCN_BGOFF;
+
+        switchToHIRCD4();
+
+        /* Set SIMO clock, VDDCSW, VregO_B voltage for DeepSleep */
+        /*void MXC_LP_SIMOprepForDeepSleep(uint32_t voltage)*/
+        {
+            /* Prevent SIMO soft start on wakeup */
+            MXC_LP_FastWakeupDisable();
+
+            /* Enable VDDCSWEN=1 prior to enter DEEPSLEEP */
+            MXC_MCR->ctrl |= MXC_F_MCR_CTRL_VDDCSWEN;
+
+            /* SIMO clock setup for deep sleep */
+            *(volatile int *)0x40005434 = 1; /* SIMOCLKDIV [1:0] : 0=div1; 1=div8; 2=div1 3=div16 */
+            /* BUCK_CLKSEL [25:24] : 0=8K; 1=16K; 2=30K; 3=RFU */
+            *(volatile int *)0x40005440 = (*(volatile int *)0x40005440 & (~(0x3 << 24))) |
+                                          (0x2 << 24);
+            /* BUCK_CLKSEL_LP [7:6] : 0=8K; 1=16K; 2=30K; 3=RFU */
+            *(volatile int *)0x40005444 = (*(volatile int *)0x40005444 & (~(0x3 << 6))) |
+                                          (0x2 << 6);
```

Comment:
Should these register locations be brought out to the register files? We can add them to pwrseq_regs.h/pwrseq_me14.svd if these are missing registers that should be included.

Reply:
Not sure. I have no objections to adding registers. However there must be a reason why those registers were not added in the first place. I suspect those are "hidden" registers that customers should not mess with.

Reply:
Okay, I'll check with Hiep.